src/ir/access_path_selectors.cc

```diff
@@ -0,0 +1,23 @@
+#include "src/ir/selector.h"
+
+#include <string>
+
+#include "absl/strings/str_join.h"
+#include "src/ir/access_path_selectors.h"
+
+namespace raksha::ir {
+
+// We create the AccessPathSelectors string by just joining the contents of the
+// component vector in reverse, using '.' as a separator.
+std::string AccessPathSelectors::ToString() const {
```

Comment:
This could move into the header if you can use an inline lambda expression unless you plan to add more functionality to this file.